### PR TITLE
Add support for publishing scoop manifests

### LINKED ISSUE #37: Add support for scoop installation manager for Windows systems
Since Homebrew doesn't support Windows systems the best equivalent option for Windows is [scoop.sh](https://scoop.sh). 
Supporting **Brew** and **Scoop** git repo updates allows publishing an application to any target system.

### LINKED ISSUE #45: Release assets should be always uploaded
`UpdateChangelogAndUploadPackages` doesn't upload a new release asset if an old one with the same name is already attached to a release.

While implementing #37, I noticed that the new asset and the old asset have different hashes.
This breaks generating the scoop manifest since the manifest is updated using the asset of the new asset.

### PR DESCRIPTION
**Pull request by dwenegar**
Closes #37
Closes #45 

Add support for publishing Scoop manifests in a way similar to how it's done for Homebrew formulas.
Scoop provides a repository template for new buckets, so I updated the Oktokit dependency to the latest version in order to be able to create the new bucket using that template.

The original behavior of the `publish` command is to skip uploaded release assets if one asset with the same name is already attached to the release being updated. 
This causes the subsequent generation of the scoop manifest to use a SHA256 of the assets that's different from the one attached to the release making the manifest invalid.

To remedy this problem I added a `--force-upload` option to the `publish` command: when not specified the command will fail if there are already uploaded assets.

The documentation has been updated accordingly.

### CODE REVIEW

File: src/dotnet-releaser/DevHosting/GitHubDevHosting.cs [hunk 1]
```diff
@@ -466,6 +468,68 @@ public async Task UploadHomebrewFormula(string user, string repo, ProjectPackage
...
+        var shouldCreate = repositoryContents is null || repositoryContents.Count == 0;
+        if (shouldCreate)
+        {
+            _log.Info($"Creating Scoop manifest {user}/{repo}/{filePath}");
+            await _client.Repository.Content.CreateFile(user, repo, filePath, new CreateFileRequest($"{projectPackageInfo.Version}", scoopManifest));
+        }
+        else
+        {
+            Debug.Assert(repositoryContents is not null);
+            var file = repositoryContents[0];
```
Comment (xoofx) [inline src/dotnet-releaser/DevHosting/GitHubDevHosting.cs:519]: How is it that the first content is always going to be the manifest?

Comment (dwenegar) [reply src/dotnet-releaser/DevHosting/GitHubDevHosting.cs]: Totally missed that.

File: src/dotnet-releaser/DevHosting/GitHubDevHosting.cs [hunk 2]
```diff
@@ -349,18 +351,18 @@ public async Task UpdateChangelogAndUploadPackages(string user, string repo, Rel
             }
 
             var filename = Path.GetFileName(entry.Path);
-            if (assets.Any(x => x.Name == filename))
+            if (assets.Any(x => x.Name == filename) && !forceUpload)
             {
-                _log.Info($"No need to update {entry.Path} on GitHub. Already uploaded.");
-                continue;
+                _log.Error($"{entry.Path} has already been uploaded. Use --force-upload to replace it.");
+                return;
```
Comment (xoofx) [inline src/dotnet-releaser/DevHosting/GitHubDevHosting.cs:357]:
```suggestion
                _log.Warning($"{entry.Path} has already been uploaded. Use --force-upload to replace it.");
                continue;
```

Comment (xoofx) [reply src/dotnet-releaser/DevHosting/GitHubDevHosting.cs:357]: Keep a warning to not break existing workflow that were relying on this.

File: src/dotnet-releaser/DevHosting/GitHubDevHosting.cs [hunk 3]
```diff
@@ -349,9 +351,9 @@ public async Task UpdateChangelogAndUploadPackages(string user, string repo, Rel
             }
 
             var filename = Path.GetFileName(entry.Path);
-            if (assets.Any(x => x.Name == filename))
+            if (assets.Any(x => x.Name == filename) && !forceUpload)
             {
-                _log.Info($"No need to update {entry.Path} on GitHub. Already uploaded.");
+                _log.Warning($"{entry.Path} has already been uploaded. Use --force-upload to replace it.");
```
Comment (xoofx) [inline src/dotnet-releaser/DevHosting/GitHubDevHosting.cs:356]:
```suggestion
                _log.Info($"{entry.Path} has already been uploaded. Use --force-upload to replace it.");
```

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**